### DISCUSSION TITLE
Are there plans to eventually support JSON_TABLE?

### DISCUSSION THREAD

**Discussion by jamespohalloran** [planetscale/discussion#351] [Feature Requests] [answered]
Really liking PlanetScale so far! But a bit stuck as my application relies on `JSON_TABLE` for a pretty important query. Is that something in the roadmap to be supported later? 

--
For context of what I'm using it for..

(Maybe there's a way to do what I'm doing without JSON_TABLE.
TLDR; I'm grouping some queries by events that are in a nearby daterange of eachother. Finding it tricky to do much array manipulation in mysql without JSON_TABLE:

```
/...
LEFT JOIN User
    ON UserOnPerformers.userId = User.id
CROSS JOIN json_table(concat('[',EXTRACT(WEEK FROM datetime_local),',',EXTRACT(WEEK FROM DATE_ADD(datetime_local, INTERVAL IF(dayofweek(datetime_local) < 3,-4,3) DAY )), ']')
                    , '$[*]' COLUMNS(
                               week VARCHAR(50) PATH '$'
                             )
) AS timeframes
```

**Comment by orware** [comment, marked as answer]
@jamespohalloran,

This is definitely on the todo list within Vitess and it's larger [MySQL Compatibility](https://github.com/vitessio/vitess/projects/4#card-33372622) plan and we have also touched on the `JSON_TABLE` limitation within our own [PlanetScale MySQL Compatibility](https://planetscale.com/docs/reference/mysql-compatibility) documentation page. 

In the meanwhile, the closest I came across to a potential workaround to using `JSON_TABLE` was described in an answer to this [Stack Overflow question](https://stackoverflow.com/questions/69181190/is-there-any-alternative-to-json-table), but it would have to be adapted to work with your specific schema. 

Another potential tip I can think of might be to utilize [generated columns from the JSON and indexing the results](https://planetscale.com/blog/indexing-json-in-mysql).

**Reply by jamespohalloran** [reply]
Awesome ,thank you! I did end up finding a workaround using a table of static rows, but this would be a great addition in the future. 
Keep up the great work team!
